rfcs/0000-light-rail-trip-model.md

```diff
@@ -0,0 +1,225 @@
+
+The Glides application will be primarily responsible for establishing the train - trip matching. While exact details of the logic are beyond the scope of this RFC, factors that are likely to be weighed include:
+
+1. Cars entered for trips on trainsheets
+1. Additional mid-line edits to trips by Glides users, however those are represented
+1. Train movement (for instance, marking a train as no longer assigned to a trip when it reaches that trip's endpoint, or assigning it to a different trip if appropriate)
+1. Train location (for instance, marking a train as no longer assigned to a trip if its current location is no longer on the path from the start point to the end point, including being headed in the correct direction)
+1. AVI code (indicating train destination)
+
+### Required properties of Glides-produced trip assignments
+
+While there are not hard requirements around this, Glides SHOULD generally try to produce trip assignments for any vehicle tracking in the realtime data that is operating on the mainline (as opposed to in a yard). This is important for RTR to be able to make predictions without doing additional trip assignment logic (any train operating in service should be getting predictions, whether revenue train predictions in the public GTFS-rt feed or non-revenue predictions for in-station announcements about trains that won't stop). Additionally, it has benefits for operations (having a trip in the internal Glides data model provides a location to store any additional metadata, such as notes). For the time being, trip matches performed to provide a placeholder trip for a train not associated with a trip explicitly via the cars entered on a trainsheet will generally be ADDED trips that do not appear on the trainsheet interface. They will, however, generate corresponding `com.mbta.ctd.glides.trips_updated` events creating those added trips.
+
+Once this feed is implemented, RTR MUST adopt it as its source of truth for determining trip assignments in the data that it publishes. If RTR receives a `vehicle_trip_assignment` event for a trip ID that it does not recognize (via either the GTFS schedule or via a trip add `trips_updated` event), it SHOULD map that to a unique added trip ID of its own and maintain that mapping in the GTFS-rt feed for the duration of the trip (that is, the unrecognized trip ID from Glides should not suddenly become mapped to a different randomly-generated added trip). This is to handle the case where RTR is aware of a different set of trip IDs due to a disruption modeled in `gtfs_creator` but not in HASTUS. See [Rationale and alternatives](#rationale-and-alternatives) for some brief discussion of the tradeoff involved here.
```

Comment:
Does this also suggest that RTR should send back to Glides a mapping of the new trip IDs so that, for instance, Glides could keep its records straight in the playback map tool and maybe the ladder? Or maybe it's fine if Glides can just align based on the vehicle ID?

Reply:
Good question - I hadn't really considered playback but I'm not sure that I really know enough about our requirements around that to say one way or another. As far as the ladder goes, my assumption is that since Glides would be keeping track of the vehicle ID - trip matching on its own end, it could use that (unless I'm missing some edge case where we would _only_ have RTR-assigned trip ID on some data that we want to associate with the ladder).